presto-main/src/test/java/com/facebook/presto/operator/scalar/TestArrayFindFunction.java

```diff
@@ -0,0 +1,92 @@
+import static java.util.Arrays.asList;
+
+public class TestArrayFindFunction
+        extends AbstractTestFunctions
+{
+    @Test
+    public void testBasic()
+    {
+        assertFunction("find(ARRAY [5, 6], x -> x = 5)", INTEGER, 5);
+        assertFunction("find(ARRAY [BIGINT '5', BIGINT '6'], x -> x = 5)", BIGINT, 5L);
+        assertFunction("find(ARRAY [5, 6], x -> x > 5)", INTEGER, 6);
+        assertFunction("find(ARRAY [null, false, true, false, true, false], x -> nullif(x, false))", BOOLEAN, true);
+        assertFunction("find(ARRAY [null, true, false, null, true, false, null], x -> not x)", BOOLEAN, false);
+        assertFunction("find(ARRAY [4.8E0, 6.2E0], x -> x > 5)", DOUBLE, 6.2);
+        assertFunction("find(ARRAY ['abc', 'def', 'ayz'], x -> substr(x, 1, 1) = 'a')", createVarcharType(3), "abc");
+        assertFunction(
+                "find(ARRAY [ARRAY ['abc', null, '123'], ARRAY ['def', 'x', '456']], x -> x[2] IS NULL)",
+                new ArrayType(createVarcharType(3)),
+                asList("abc", null, "123"));
+    }
+
+    @Test
+    public void testPositiveOffset()
+    {
+        assertFunction("find(ARRAY [5, 6], 2, x -> x = 5)", INTEGER, null);
+        assertFunction("find(ARRAY [5, 6], 4, x -> x > 0)", INTEGER, null);
+        assertFunction("find(ARRAY [5, 6, 7, 8], 3, x -> x > 5)", INTEGER, 7);
+        assertFunction("find(ARRAY [null, false, true, null, true, false], 4, x -> nullif(x, false))", BOOLEAN, true);
+        assertFunction("find(ARRAY [4.8E0, 6.2E0, 7.8E0], 3, x -> x > 5)", DOUBLE, 7.8);
+        assertFunction("find(ARRAY ['abc', 'def', 'ayz'], 2, x -> substr(x, 1, 1) = 'a')", createVarcharType(3), "ayz");
+        assertFunction(
+                "find(ARRAY [ARRAY ['abc', null, '123'], ARRAY ['def', null, '456']], 2, x -> x[2] IS NULL)",
+                new ArrayType(createVarcharType(3)),
+                asList("def", null, "456"));
+    }
+
+    @Test
+    public void testNegativeOffset()
+    {
+        assertFunction("find(ARRAY [5, 6], -2, x -> x > 5)", INTEGER, null);
+        assertFunction("find(ARRAY [5, 6], -4, x -> x > 0)", INTEGER, null);
+        assertFunction("find(ARRAY [5, 6, 7, 8], -2, x -> x > 5)", INTEGER, 7);
+        assertFunction("find(ARRAY [9, 6, 3, 8], -2, x -> x > 5)", INTEGER, 6);
+        assertFunction("find(ARRAY [null, false, true, null, true, false], -3, x -> nullif(x, false))", BOOLEAN, true);
+        assertFunction("find(ARRAY [4.8E0, 6.2E0, 7.8E0], -2, x -> x > 5)", DOUBLE, 6.2);
+        assertFunction("find(ARRAY ['abc', 'def', 'ayz'], -2, x -> substr(x, 1, 1) = 'a')", createVarcharType(3), "abc");
+        assertFunction(
+                "find(ARRAY [ARRAY ['abc', null, '123'], ARRAY ['def', null, '456']], -2, x -> x[2] IS NULL)",
+                new ArrayType(createVarcharType(3)),
+                asList("abc", null, "123"));
+    }
+
+    @Test
+    public void testEmpty()
+    {
```

Comment:
Add a test where array is null. Like find(null, x -> true)

Reply:
Added test.